pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -305,28 +305,38 @@ func (s *Scheduler) Solve(ctx context.Context, pods []*corev1.Pod) (Results, err
 		if !ok {
 			break
 		}
-
 		// Schedule to existing nodes or create a new node
-		if podErrors[pod] = s.add(ctx, pod); podErrors[pod] == nil {
-			delete(podErrors, pod)
+		p := pod.DeepCopy()
+		for {
```

Comment:
Thoughts on breaking this out into a separate function?

Reply:
Good call -- added the change